sway-types/src/source_engine.rs

```diff
@@ -0,0 +1,46 @@
+use std::{collections::HashMap, path::PathBuf, sync::RwLock};
+
+use crate::SourceId;
+
+#[derive(Debug, Default)]
+pub struct SourceEngine {
```

Comment:
I think it might be worth it to add some comments that explain what the source engine does and why the internals are RwLocked.